### PR TITLE
test: fix benchmark tests

### LINKED ISSUE #208: Bug FaissPostgress - int not parsed
When defining the pod via yaml, the shards and replicas are interpreted as string.
This leads to the exception bellow:

Config
```
- name: text_indexer
    shards: 1
    uses: 'jinahub+docker://FaissPostgresSearcher'
    uses_with:
      startup_sync_args:
        only_delta: true
      total_shards: shards
      hostname: postgres.postgres.svc.cluster.local
      username: postgresadmin
      database: postgresdb
      table: text_indexer
```

Error
```
File "/usr/local/lib/python3.7/site-packages/jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py", line 129, in __init__
    self._load_dump(dump_path, dump_func, prefetch_size, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/jinahub/indexers/searcher/FaissSearcher/faiss_searcher.py", line 145, in _load_dump
    iterator = dump_func(shard_id=self.runtime_args.pea_id)
  File "/usr/local/lib/python3.7/site-packages/jinahub/indexers/storage/PostgreSQLStorage/postgres_indexer.py", line 264, in _get_delta
    shard_id, total_shards, self.virtual_shards
  File "/usr/local/lib/python3.7/site-packages/jinahub/indexers/storage/PostgreSQLStorage/postgres_indexer.py", line 239, in _vshards_to_get
    if shard_id > total_shards - 1:
TypeError: unsupported operand type(s) for -: 'str' and 'int'

```

### PR DESCRIPTION
**Pull request by cristianmtr**
Various fixes for the PSQLFaiss benchmarks and tests

Also fixes #208

### CODE REVIEW

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py [hunk 1]
```diff
@@ -39,19 +39,22 @@ def __init__(
         self.logger = get_logger(self)
 
         if total_shards is None:
-            self.total_shards = getattr(self.runtime_args, 'parallel', None)
+            self.total_shards = getattr(self.runtime_args, "parallel", None)
```
Comment (maximilianwerk) [inline jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py:42]: seems like you missed the `-no-string-format` flag for black. Can you revert this?

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py [hunk 2]
```diff
@@ -62,29 +65,29 @@ def __init__(
...
         if startup_sync_args:
-            startup_sync_args['startup'] = True
+            startup_sync_args["startup"] = True
             self.sync(parameters=startup_sync_args)
 
     def _init_executors(self, dump_path, kwargs, startup_sync_args):
         # float32 because that's what faiss expects
         kv_indexer = PostgreSQLStorage(dump_dtype=np.float32, **kwargs)
-        vec_indexer = FaissSearcher(dump_path=dump_path, **kwargs)
+        vec_indexer = FaissSearcher(dump_path=dump_path, prefetch_size=12, **kwargs)
```
Comment (maximilianwerk) [inline jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py:74]: Why 12? Sounds quite arbitrary.

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/tests/unit/test_unit.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+d_embedding = np.array([1, 1, 1, 1, 1, 1, 1])
+c_embedding = np.array([2, 2, 2, 2, 2, 2, 2])
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+compose_yml = os.path.join(cur_dir, '..', 'docker-compose.yml')
+
+
+@pytest.mark.parametrize('docker_compose', [compose_yml], indirect=['docker_compose'])
+def test_shards_str(docker_compose):
```
Comment (maximilianwerk) [inline jinahub/indexers/searcher/compound/FaissPostgresSearcher/tests/unit/test_unit.py:33]: I would argue, anything that uses a docker-compose is not a unit test anymore and thus have not the renaming to `unit/test_unit.py`.

File: tests/requirements.txt [hunk 4]
```diff
@@ -1,4 +1,4 @@
-git+https://github.com/jina-ai/jina.git@v2.0.21#egg=jina[standard]
+git+https://github.com/jina-ai/jina.git@b0e46a3027a24b7e251085c13953b2ad38034a53#egg=jina[standard] # includes a fix for handling DA.extend(None)
```
Comment (maximilianwerk) [inline tests/requirements.txt:1]: Can this be changed back to a proper Jina version again?

**Review by maximilianwerk — CHANGES_REQUESTED** [review]
I am not super happy about the non-executor specific integration tests. Now, that we have the FaissPostgres combination we might want to outsource all the dump-reload mechanism testing towards only this indexer soon. (not in this PR)

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py (hunk 2 above)
Comment (cristianmtr) [reply jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py:74]: 512 was the default. I just needed a smaller number. Idk @numb3r3 ?

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/tests/unit/test_unit.py (hunk 3 above)
Comment (cristianmtr) [reply jinahub/indexers/searcher/compound/FaissPostgresSearcher/tests/unit/test_unit.py:33]: it depends on docker for the PSQL db. Can't really test PSQL without a database. I'll move it to the integration test folder

File: tests/requirements.txt (hunk 4 above)
Comment (cristianmtr) [reply tests/requirements.txt]: done

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py (hunk 1 above)
Comment (cristianmtr) [reply jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py]: fixed

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py (hunk 2 above)
Comment (maximilianwerk) [reply jinahub/indexers/searcher/compound/FaissPostgresSearcher/faisspsql.py:74]: take some power of 2.

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/tests/integration/test_208.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import os
```
Comment (maximilianwerk) [inline jinahub/indexers/searcher/compound/FaissPostgresSearcher/tests/integration/test_208.py:1]: Not obvious why `test_208.py`!? I want to know the background story! Some smear campaign involved?

**Review by maximilianwerk — APPROVED** [review]
LGTM

File: jinahub/indexers/searcher/compound/FaissPostgresSearcher/tests/integration/test_208.py (hunk 5 above)
Comment (cristianmtr) [reply jinahub/indexers/searcher/compound/FaissPostgresSearcher/tests/integration/test_208.py:1]: > Also fixes #208

**Review by maximilianwerk — APPROVED** [review]
LGTM

**Pull request merged**